pgxn/neon/libpagestore.c

```diff
@@ -454,12 +454,12 @@ pg_init_libpagestore(void)
 							NULL, NULL, NULL);
 	DefineCustomIntVariable("neon.readahead_buffer_size",
 							"number of prefetches to buffer",
-							"This buffer is used to store prefetched data; so "
-							"it is important that this buffer is at least as "
-							"large as the configured value of all tablespaces' "
-							"effective_io_concurrency and maintenance_io_concurrency, "
-							"your sessions' values of these, and the value for "
-							"seqscan_prefetch_buffers.",
+							"This buffer is used to hold and manage prefetched "
+							"data; so it is important that this buffer is at "
+							"least as large as the configured value of all "
+							"tablespaces' effective_io_concurrency and "
+							"maintenance_io_concurrency, and your sessions' "
+							"values for these settings.",
 							&readahead_buffer_size,
```

Comment:
It is still not clear to me to me why
1, min value is 16 (and not 1 or 0)
2. default value is 16 and not 200-300 as recommended for SSD

Reply:
What happens if you have multiple concurrent sequential scans, each doing prefetching on their own? If effective_io_concurrency=64, and neon.readahead_buffer_size=128, I guess that means that you can fit two concurrent sequential scans in the buffers. How does the system behave if you have three concurrent sequential scans in the same backend? Does it degrade gracefully, or start to discard prefetched responses and have performance fall off a cliff?

Having multiple concurrent large sequential scans is not common, I'm not sure what the plan would even look like.  But I believe we issue prefetches even for fairly small tables, so that could also happen with a sequential scan of large table, with a nested loop join of a seqscan of a relatively small table. Or if you have a function in the query that performs a sequential scan.

I believe that's a pre-existing issue, so doesn't need to be addressed by this PR. But perhaps set the default `neon.readahead_buffer_size` to much larger value, e.g. 1024, to stay further away from the edge of the cliff.
 There's not much penalty from having a larger-than-necessary `neon.readahead_buffer_size` setting, right? Just a a few kilobytes of wasted memory.

Reply:
> So, while empty readahead buffers would be cheap, there is no guarantee they will be empty on average; in which case the overhead is effectively much more than just a few kilobytes.

Good point.

Reply:
It will be interesting to estimate percent of  false-positives.
From one side, postgres is using seqscan only when it really expects large number of traversed records.
Certainly it can happen that:
- there are no indexes and so seqscan is the only possible option
- seqscan is terminated earlier than expected
but usually seqscan should benefit from prefetch.

What makes me concern is that test_bulk_upload shows that prefetch significantly slowdown updates. May be it will be fixed by last written lsn patch.